build-support/php: refactor build scripts

drupol opened the pull request:
Continuation of #385830

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

drupol committed:
build-support/php: refactor build scripts

drupol committed:
phpPackages.php-cs-fixer: 3.67.0 -> 3.70.1

drupol commented:
@ShamrockLee I've been working on this and refactoring a bit more, but I'm still unable to override `vendorHash` using `<derivation>.overrideAttrs { vendorHash = "foo"; };`

 Do you mind having a look and let me know any improvements?

ShamrockLee commented on pkgs/build-support/php/builders/v2/build-composer-vendor.nix:
```diff
@@ -8,14 +8,14 @@ let
-      composerNoPlugins ? true,
-      composerNoScripts ? true,
-      composerStrictValidation ? true,
+      php ? finalAttrs.php or toplevel.php,
+      composer ? finalAttrs.php.packages.composer or toplevel.php.packages.composer,
+      composerLock ? finalAttrs.composerLock or null,
+      vendorHash ? finalAttrs.vendorHash or "",
+      composerNoDev ? finalAttrs.composerNoDev or true,
+      composerNoPlugins ? finalAttrs.composerNoPlugins or true,
+      composerNoScripts ? finalAttrs.composerNoScripts or true,
+      composerStrictValidation ? finalAttrs.composerStrictValidation or true,
```
`{ <argname> ? finalAttrs.<argname> or <default> }: { }` makes no sense to me, TBH. `{ <argname> ? <default> }` should work.

drupol replied:
You're right, I fixed it.

drupol replied:
Yes, that was the idea. Is it OK if I restore it ?

ShamrockLee replied:
I think so.

drupol replied:
done :)

piotrkwiecinski commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 386757`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>php81Packages.psalm</li>
  </ul>
</details>
<details>
  <summary>:x: 23 packages failed to build:</summary>
  <ul>
    <li>laravel</li>
    <li>paratest</li>
    <li>pest</li>
    <li>phel</li>
    <li>php81Packages.box (php82Packages.box, php83Packages.box, php84Packages.box)</li>
    <li>php81Packages.deployer</li>
    <li>php81Packages.phing</li>
    <li>php81Packages.phpinsights</li>
    <li>php81Packages.psysh</li>
    <li>php82Packages.deployer</li>
    <li>php82Packages.phing</li>
    <li>php82Packages.phpinsights</li>
    <li>php82Packages.psysh</li>
    <li>php83Packages.deployer</li>
    <li>php83Packages.phing</li>
    <li>php83Packages.phpinsights</li>
    <li>php83Packages.psysh</li>
    <li>php84Packages.deployer</li>
    <li>php84Packages.phing</li>
    <li>php84Packages.phpinsights</li>
    <li>php84Packages.psysh</li>
    <li>phpdocumentor</li>
    <li>robo</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 54 packages built:</summary>
  <ul>
    <li>agorakit</li>
    <li>composer-require-checker</li>
    <li>davis</li>
    <li>flarum</li>
    <li>grocy</li>
    <li>invoiceplane</li>
    <li>kimai</li>
    <li>librenms</li>
    <li>movim</li>
    <li>n98-magerun</li>
    <li>n98-magerun2</li>
    <li>pdepend</li>
    <li>php81Packages.castor</li>
    <li>php81Packages.grumphp</li>
    <li>php81Packages.phive</li>
    <li>php81Packages.php-codesniffer</li>
    <li>php81Packages.php-cs-fixer</li>
    <li>php81Packages.php-parallel-lint</li>
    <li>php81Packages.phpmd</li>
    <li>php81Packages.phpstan</li>
    <li>php82Packages.castor</li>
    <li>php82Packages.grumphp</li>
    <li>php82Packages.phive</li>
    <li>php82Packages.php-codesniffer</li>
    <li>php82Packages.php-cs-fixer</li>
    <li>php82Packages.php-parallel-lint</li>
    <li>php82Packages.phpmd</li>
    <li>php82Packages.phpstan</li>
    <li>php82Packages.psalm</li>
    <li>php83Packages.castor</li>
    <li>php83Packages.grumphp</li>
    <li>php83Packages.phive</li>
    <li>php83Packages.php-codesniffer</li>
    <li>php83Packages.php-cs-fixer</li>
    <li>php83Packages.php-parallel-lint</li>
    <li>php83Packages.phpmd</li>
    <li>php83Packages.phpstan</li>
    <li>php83Packages.psalm</li>
    <li>php84Packages.castor</li>
    <li>php84Packages.grumphp</li>
    <li>php84Packages.phive</li>
    <li>php84Packages.php-codesniffer</li>
    <li>php84Packages.php-cs-fixer</li>
    <li>php84Packages.php-parallel-lint</li>
    <li>php84Packages.phpmd</li>
    <li>php84Packages.phpstan</li>
    <li>php84Packages.psalm</li>
    <li>phpactor</li>
    <li>phpunit</li>
    <li>pixelfed</li>
    <li>roave-backward-compatibility-check</li>
    <li>simplesamlphp</li>
    <li>snipe-it</li>
    <li>vimPlugins.phpactor</li>
  </ul>
</details>

piotrkwiecinski commented:
deployer, phing, psysh, phpinsights have different hash.

box, phpdocumentor fail due to dangling symlink.

drupol commented:
> deployer, phing, psysh, phpinsights, paratest, pest have different hash.
> 
> box, phpdocumentor fail due to dangling symlink.

Going to investigate at the end of the day.

piotrkwiecinski commented:
Dangling symlinks aren't related to this PR so they should be fixed separately.

drupol commented:
> deployer, phing, psysh, phpinsights, paratest, pest have different hash.

This is worrying me. I will first investigate later on why this.

drupol committed:
pest: 3.3.1 -> 3.7.4

drupol committed:
deployer: 7.4.0 -> 7.5.12-unstable-2025-03-02

The release management of deployer is something I've never seen.
They basically commit the PHAR file in the source, in a specific branch.
They also modify the runner `bin/dep` to use that PHAR. Therefore, this
break on NixOS since we are using the PHAR.
A workaround is to use the master branch that doesn't contain the PHAR.

drupol committed:
phpPackages.phing: update hash, add `installCheck`

drupol committed:
phpPackages.psysh: update hash, add `installCheck`

drupol committed:
phpPackages.phpinsights: update hash, add `installCheck`

drupol committed:
paratest: 7.6.0 -> 7.8.2

drupol committed:
phpdocumentor: 3.6.0 -> 3.7.1

piotrkwiecinski commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 386757`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>php81Packages.psalm</li>
  </ul>
</details>
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>laravel</li>
    <li>phel</li>
    <li>php81Packages.deployer</li>
    <li>php81Packages.phpinsights</li>
    <li>php82Packages.deployer</li>
    <li>php83Packages.deployer</li>
    <li>php84Packages.deployer</li>
    <li>robo</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 69 packages built:</summary>
  <ul>
    <li>agorakit</li>
    <li>composer-require-checker</li>
    <li>davis</li>
    <li>flarum</li>
    <li>grocy</li>
    <li>invoiceplane</li>
    <li>kimai</li>
    <li>librenms</li>
    <li>movim</li>
    <li>n98-magerun</li>
    <li>n98-magerun2</li>
    <li>paratest</li>
    <li>pdepend</li>
    <li>pest</li>
    <li>php81Packages.box (php82Packages.box, php83Packages.box, php84Packages.box)</li>
    <li>php81Packages.castor</li>
    <li>php81Packages.grumphp</li>
    <li>php81Packages.phing</li>
    <li>php81Packages.phive</li>
    <li>php81Packages.php-codesniffer</li>
    <li>php81Packages.php-cs-fixer</li>
    <li>php81Packages.php-parallel-lint</li>
    <li>php81Packages.phpmd</li>
    <li>php81Packages.phpstan</li>
    <li>php81Packages.psysh</li>
    <li>php82Packages.castor</li>
    <li>php82Packages.grumphp</li>
    <li>php82Packages.phing</li>
    <li>php82Packages.phive</li>
    <li>php82Packages.php-codesniffer</li>
    <li>php82Packages.php-cs-fixer</li>
    <li>php82Packages.php-parallel-lint</li>
    <li>php82Packages.phpinsights</li>
    <li>php82Packages.phpmd</li>
    <li>php82Packages.phpstan</li>
    <li>php82Packages.psalm</li>
    <li>php82Packages.psysh</li>
    <li>php83Packages.castor</li>
    <li>php83Packages.grumphp</li>
    <li>php83Packages.phing</li>
    <li>php83Packages.phive</li>
    <li>php83Packages.php-codesniffer</li>
    <li>php83Packages.php-cs-fixer</li>
    <li>php83Packages.php-parallel-lint</li>
    <li>php83Packages.phpinsights</li>
    <li>php83Packages.phpmd</li>
    <li>php83Packages.phpstan</li>
    <li>php83Packages.psalm</li>
    <li>php83Packages.psysh</li>
    <li>php84Packages.castor</li>
    <li>php84Packages.grumphp</li>
    <li>php84Packages.phing</li>
    <li>php84Packages.phive</li>
    <li>php84Packages.php-codesniffer</li>
    <li>php84Packages.php-cs-fixer</li>
    <li>php84Packages.php-parallel-lint</li>
    <li>php84Packages.phpinsights</li>
    <li>php84Packages.phpmd</li>
    <li>php84Packages.phpstan</li>
    <li>php84Packages.psalm</li>
    <li>php84Packages.psysh</li>
    <li>phpactor</li>
    <li>phpdocumentor</li>
    <li>phpunit</li>
    <li>pixelfed</li>
    <li>roave-backward-compatibility-check</li>
    <li>simplesamlphp</li>
    <li>snipe-it</li>
    <li>vimPlugins.phpactor</li>
  </ul>
</details>

piotrkwiecinski commented:
We still have:
```
error: hash mismatch in fixed-output derivation '/nix/store/5lll6bycrhjbk6hb1mi4aar4z5qbm8jx-phel-composer-vendor-0.16.1.drv':
         specified: sha256-jcMfGPnGoSP8E1JXPEqKr53amV0d08GvGfe8niwM8Q4=
            got:    sha256-SDLpl2gBvtVjREfcy1WDFqsGRK1fKr2wKPuBkPhApNI
```

robo

```
error: hash mismatch in fixed-output derivation '/nix/store/yz6y50404489d98nw3plwd8g0ax6zwwc-robo-composer-vendor-5.1.0.drv':
         specified: sha256-PYtZy6c/Z1GTcYyfU77uJjXCzQSfBaNkon8kqGyVq+o=
            got:    sha256-vketnTu5VEgt3HBbtnTppWl3+sSSIsCB2MpvL27bxv4=
```
laravel
```
error: hash mismatch in fixed-output derivation '/nix/store/cqfxlj7b019hsmjrmnp1d7sj4524pvav-laravel-composer-vendor-5.9.2.drv':
         specified: sha256-jUg0hmkShzK1CAO3+Btqe3/5GFKVxRKDtIxmUuU3EdU=
            got:    sha256-Kid5Vp42Pme1AO9fI9e/f24Cdzi1uV8qEULFEvQDKnE=
```

after changing deployer version we get:

```
error: builder for '/nix/store/wklpss0dkv6fvd8hc8yyln4h0x7s7sxi-deployer-composer-vendor-7.5.12-unstable-2025-03-02.drv' failed with exit code 1;
       last 21 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/plnwsgpf90n0fw2ns99d5w7pjch1ya2r-source
       > source root is source
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > Executing composerVendorConfigureHook
       > Setting COMPOSER_ROOT_VERSION to 7.5.12-unstable-2025-03-02
       > Using `composer.lock` file from the source package
       > Finished composerVendorConfigureHook
       > Running phase: buildPhase
       > Executing composerVendorBuildHook
       > Setting some required environment variables for Composer...
       >
       > In VersionParser.php line 191:
       >                                                        
       >   Invalid version string "7.5.12-unstable-2025-03-02" 
```

phpinsights

```
phpinsights to be relative to /nix/store/gzvd1zr1ha2lwfa64zvjd2d35i43a3bp-phpinsights-2.12.0
       > Running phase: installCheckPhase
       > Executing versionCheckPhase
       > Did not find version 2.12.0 in the output of the command /nix/store/gzvd1zr1ha2lwfa64zvjd2d35i43a3bp-phpinsights-2.12.0/bin/phpinsights --version 
```

drupol commented:
Thanks, looks like I overlooked some parts, going to give it a shot again asap.

drupol committed:
phel: update hash, add `installCheck`

drupol committed:
robo: update hash, add `installCheck`

drupol committed:
laravel: update hash

ShamrockLee commented:
> > deployer, phing, psysh, phpinsights, paratest, pest have different hash.
> 
> This is worrying me. I will first investigate later on why this.

Fetchers' change of hashing would be a sjgnificant backward incompatibility, which would hardly be remedied just by a tree-wide hash update.

Output diff across the change might help with the investigation.

drupol commented:
I'm 90% sure that the hash change comes from an old PR. I'm quite confident that the current builder is quite stable, I would not worry about this at the moment.

piotrkwiecinski commented:
I reckon it's fine to just remove invalid symlinks. Probably they should be resolved upstream. I don't think we should remove the check completely.

drupol committed:
build-support/php: enable support for `dontCheckForBrokenSymlinks` set to `true` by default

piotrkwiecinski commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 386757`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>php81Packages.psalm</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>php81Packages.phpinsights</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 76 packages built:</summary>
  <ul>
    <li>agorakit</li>
    <li>composer-require-checker</li>
    <li>davis</li>
    <li>flarum</li>
    <li>grocy</li>
    <li>invoiceplane</li>
    <li>kimai</li>
    <li>laravel</li>
    <li>librenms</li>
    <li>movim</li>
    <li>n98-magerun</li>
    <li>n98-magerun2</li>
    <li>paratest</li>
    <li>pdepend</li>
    <li>pest</li>
    <li>phel</li>
    <li>php81Packages.box (php82Packages.box, php83Packages.box, php84Packages.box)</li>
    <li>php81Packages.castor</li>
    <li>php81Packages.deployer</li>
    <li>php81Packages.grumphp</li>
    <li>php81Packages.phing</li>
    <li>php81Packages.phive</li>
    <li>php81Packages.php-codesniffer</li>
    <li>php81Packages.php-cs-fixer</li>
    <li>php81Packages.php-parallel-lint</li>
    <li>php81Packages.phpmd</li>
    <li>php81Packages.phpstan</li>
    <li>php81Packages.psysh</li>
    <li>php82Packages.castor</li>
    <li>php82Packages.deployer</li>
    <li>php82Packages.grumphp</li>
    <li>php82Packages.phing</li>
    <li>php82Packages.phive</li>
    <li>php82Packages.php-codesniffer</li>
    <li>php82Packages.php-cs-fixer</li>
    <li>php82Packages.php-parallel-lint</li>
    <li>php82Packages.phpinsights</li>
    <li>php82Packages.phpmd</li>
    <li>php82Packages.phpstan</li>
    <li>php82Packages.psalm</li>
    <li>php82Packages.psysh</li>
    <li>php83Packages.castor</li>
    <li>php83Packages.deployer</li>
    <li>php83Packages.grumphp</li>
    <li>php83Packages.phing</li>
    <li>php83Packages.phive</li>
    <li>php83Packages.php-codesniffer</li>
    <li>php83Packages.php-cs-fixer</li>
    <li>php83Packages.php-parallel-lint</li>
    <li>php83Packages.phpinsights</li>
    <li>php83Packages.phpmd</li>
    <li>php83Packages.phpstan</li>
    <li>php83Packages.psalm</li>
    <li>php83Packages.psysh</li>
    <li>php84Packages.castor</li>
    <li>php84Packages.deployer</li>
    <li>php84Packages.grumphp</li>
    <li>php84Packages.phing</li>
    <li>php84Packages.phive</li>
    <li>php84Packages.php-codesniffer</li>
    <li>php84Packages.php-cs-fixer</li>
    <li>php84Packages.php-parallel-lint</li>
    <li>php84Packages.phpinsights</li>
    <li>php84Packages.phpmd</li>
    <li>php84Packages.phpstan</li>
    <li>php84Packages.psalm</li>
    <li>php84Packages.psysh</li>
    <li>phpactor</li>
    <li>phpdocumentor</li>
    <li>phpunit</li>
    <li>pixelfed</li>
    <li>roave-backward-compatibility-check</li>
    <li>robo</li>
    <li>simplesamlphp</li>
    <li>snipe-it</li>
    <li>vimPlugins.phpactor</li>
  </ul>
</details>

piotrkwiecinski commented on pkgs/build-support/php/builders/v2/build-composer-vendor.nix:
```diff
@@ -8,33 +8,38 @@ let
       strictDeps ? true,
       doCheck ? true,
       doInstallCheck ? false,
+      dontCheckForBrokenSymlinks ? finalAttrs.dontCheckForBrokenSymlinks or true,
```
Wouldn't it be better to enable it by default?

drupol replied:
Actually, broken symlinks are something that might be something we can have since we use composer with `--no-dev` by default. This is why I disable it by default.

drupol committed:
phpPackages.box: 4.6.2 -> 4.6.6, fix `installCheck`

piotrkwiecinski commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 386757`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>php81Packages.phpinsights</li>
    <li>php81Packages.psalm</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 76 packages built:</summary>
  <ul>
    <li>agorakit</li>
    <li>composer-require-checker</li>
    <li>davis</li>
    <li>flarum</li>
    <li>grocy</li>
    <li>invoiceplane</li>
    <li>kimai</li>
    <li>laravel</li>
    <li>librenms</li>
    <li>movim</li>
    <li>n98-magerun</li>
    <li>n98-magerun2</li>
    <li>paratest</li>
    <li>pdepend</li>
    <li>pest</li>
    <li>phel</li>
    <li>php81Packages.box (php82Packages.box, php83Packages.box, php84Packages.box)</li>
    <li>php81Packages.castor</li>
    <li>php81Packages.deployer</li>
    <li>php81Packages.grumphp</li>
    <li>php81Packages.phing</li>
    <li>php81Packages.phive</li>
    <li>php81Packages.php-codesniffer</li>
    <li>php81Packages.php-cs-fixer</li>
    <li>php81Packages.php-parallel-lint</li>
    <li>php81Packages.phpmd</li>
    <li>php81Packages.phpstan</li>
    <li>php81Packages.psysh</li>
    <li>php82Packages.castor</li>
    <li>php82Packages.deployer</li>
    <li>php82Packages.grumphp</li>
    <li>php82Packages.phing</li>
    <li>php82Packages.phive</li>
    <li>php82Packages.php-codesniffer</li>
    <li>php82Packages.php-cs-fixer</li>
    <li>php82Packages.php-parallel-lint</li>
    <li>php82Packages.phpinsights</li>
    <li>php82Packages.phpmd</li>
    <li>php82Packages.phpstan</li>
    <li>php82Packages.psalm</li>
    <li>php82Packages.psysh</li>
    <li>php83Packages.castor</li>
    <li>php83Packages.deployer</li>
    <li>php83Packages.grumphp</li>
    <li>php83Packages.phing</li>
    <li>php83Packages.phive</li>
    <li>php83Packages.php-codesniffer</li>
    <li>php83Packages.php-cs-fixer</li>
    <li>php83Packages.php-parallel-lint</li>
    <li>php83Packages.phpinsights</li>
    <li>php83Packages.phpmd</li>
    <li>php83Packages.phpstan</li>
    <li>php83Packages.psalm</li>
    <li>php83Packages.psysh</li>
    <li>php84Packages.castor</li>
    <li>php84Packages.deployer</li>
    <li>php84Packages.grumphp</li>
    <li>php84Packages.phing</li>
    <li>php84Packages.phive</li>
    <li>php84Packages.php-codesniffer</li>
    <li>php84Packages.php-cs-fixer</li>
    <li>php84Packages.php-parallel-lint</li>
    <li>php84Packages.phpinsights</li>
    <li>php84Packages.phpmd</li>
    <li>php84Packages.phpstan</li>
    <li>php84Packages.psalm</li>
    <li>php84Packages.psysh</li>
    <li>phpactor</li>
    <li>phpdocumentor</li>
    <li>phpunit</li>
    <li>pixelfed</li>
    <li>roave-backward-compatibility-check</li>
    <li>robo</li>
    <li>simplesamlphp</li>
    <li>snipe-it</li>
    <li>vimPlugins.phpactor</li>
  </ul>
</details>

drupol commented:
Good for you @piotrkwiecinski ?

drupol commented:
Backport @ https://github.com/NixOS/nixpkgs/pull/391063
